portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceHelper.java

```diff
@@ -82,10 +82,11 @@ public String getFriendlyURL(
 		friendlyURL = getFriendlyURL(friendlyURL);
```

Comment:
Considering what getFriendlyURL() is, is there a point to normalizing twice?

Reply:
`getFriendlyURL()` is also responsible for editing the friendly URL. So the friendly URL is based on the `name` variable when we are creating a new page, but when we are editing the page's friendly URL we normalize `friendlyURL` which is the new value for friend URL the user set.

Reply:
I'm assuming this comment was meant for the other conversation, but that's fair.